src/ui/public/modals/confirm_modal.js

```diff
@@ -23,9 +27,15 @@ module.factory('confirmModal', function ($rootScope, $compile) {
   return function confirmModal(message, customOptions) {
     const defaultOptions = {
       onCancel: noop,
-      cancelButtonText: 'Cancel'
+      cancelButtonText: 'Cancel',
+      onClose: noop,
```

Comment:
I think that in every case, `onCancel` and `onClose` will reference the same handler (assuming onCancel is a handler they need in the first place).

This also introduces a gotcha where a user can forget to add an `onCancel` handler even in situations where they DID add an `onCancel`, but the dialog still works and now they have a silent error condition.

3 handlers on a dialog with 2 buttons seems strange. I think when a user clicks the close button, they mean to do the same thing as when they click the cancel button, so calling `onCancel` on close seems appropriate, and leaving this at 2 handlers seems right.